docs/rfc/0003-importer-interface.md

```diff
@@ -0,0 +1,33 @@
+# Importer Interface RFC
+
+- Contribution Name: Importer Interface/Plugin Framework
+- Implementation Owner: Ganesh Vanahalli
+
+## Problem Statement
+
+Users would like to choose how to fetch blocks, either by downloading from an algod rest endpoint or directly from the gossip network or read blocks from a file which was previously written by the exporter.
+
+The current implementation of indexer only allows downloading blocks sequentially from an algod rest endpoint starting at the round number of recently written block to postgres db and users cannot choose a specific block to download.
+
+## Proposal
+
+Importer interface allows users to fetch any particular block either from algod rest endpoint (or directly from the network or from a file written to by the exporter plugin).
+
+### Plugin Interface
+Importer plugins that are native to the Indexer (maintained within the Indexer repository) will each implementation the importer interface:
+
+```GO
+package importers
+
+// Importer defines the interface for importer plugins
+type Importer interface {
```

Comment:
The other plugin definitions have some useful methods I think we're missing here.

`Metadata() MetaData`  
This is just a convenience at this point--centralizing plugin name info, description, deprecation data, etc. In the future the idea is for the Conduit framework to use this data.

`Init(cfg Config) error`  
Your  algod importer implementation is just getting constructed with some algod client, but we should be constructing that in the Init function. The config should just be an object passed in with all the required params. This is I guess similar to your `RegisterAlgodImporter` function, but it just standardizes these across different implementations.

`Close() error`  
This is for closing network connections, files, flushing buffers, etc. Not sure if this is just a no-op for most importers, but we still probably want it.